webui/react/src/components/kit/utils/loadable.ts

```diff
@@ -0,0 +1,243 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+const exhaustive = (v: never): never => v;
+
+type MatchArgs<T, U> =
+  | {
+      Loaded: (data: T) => U;
+      NotLoaded: () => U;
+    }
+  | {
+      Loaded: (data: T) => U;
+      _: () => U;
+    }
+  | {
+      NotLoaded: () => U;
+      _: () => U;
+    };
+
+class Loadable_<T> {
```

Comment:
You don't have to do it in this PR but can we at least make a ticket to add tests for loadable?

Reply:
Done, WEB-1711